packages/ai/src/Button.ts

```diff
@@ -58,13 +82,195 @@ class Button extends UI5Element {
 	@property({ type: Boolean })
 	disabled!: boolean;
 
+	/**
+	 * Defines the current state of the component.
+	 *
+	 * **Note:** If nothing is defined, the component will be set initially to the first defined state (if any).
+	 *
+	 * @default ""
+	 * @public
+	 */
 	@property()
-	icon!: string;
+	state?: string;
 
-	@property()
-	endIcon!: string;
+	/**
+	 * Keeps the current state of the component.
+	 * @private
+	 */
+	@property({ type: Object })
+	_currentStateObject?: ButtonState;
+
+	/**
+	 * Initiates button elements fade-out phase.
+	 * @default false
+	 * @private
+	 */
+	@property({ type: Boolean })
+	fadeOut!: boolean;
+
+	/**
+	 * Initiates button fade middle phase.
+	 * @default false
+	 * @private
+	 */
+	@property({ type: Boolean })
+	fadeMid!: boolean;
+
+	/**
+	 * Initiates button elements fade-in phase.
+	 * @default false
+	 * @private
+	 */
+	@property({ type: Boolean })
+	fadeIn!: boolean;
+
+	/**
+	 * Defines the available states of the component.
+	 * **Note:** Although this slot accepts HTML Elements, it is strongly recommended that you only use `ui5-ai-button-state` components in order to preserve the intended design.
+	 * @public
+	 */
+	@slot({ type: HTMLElement, "default": true })
+	states!: Array<ButtonState>;
+
+	onBeforeRendering(): void {
+		if (this.fadeOut || this.fadeIn) {
+			return;
+		}
+
+		if (this.state && this._currentStateObject && !this._currentStateObject.name) {
+			this._currentStateObject = this._findStateByName(this.state);
+		}
+
+		const currentStateName = this._currentStateObject?.name || "";
+
+		if (!this.state) {
+			this.state = this.states.length ? this.states[0].name : "";
```

Comment:
I've made most of the requested changes except that I left `_currentStateObject` private property, because it is needed for normal function of the component - current state should be kept in order to render proper component elements (current state `icon`, `text`, `endIcon`,  and when the animation in mid state and later - the new ones) and to initiate state change animation if the new state is different than current one.